unified-scheduler-pool/src/lib.rs

```diff
@@ -1281,23 +1330,78 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
 
     #[must_use]
     fn accumulate_result_with_timings(
+        mode: SchedulingMode,
         (result, timings): &mut ResultWithTimings,
-        executed_task: HandlerResult,
-    ) -> Option<Box<ExecutedTask>> {
-        let Ok(executed_task) = executed_task else {
-            return None;
-        };
+        executed_task: Box<ExecutedTask>,
+        state_machine: &mut SchedulingStateMachine,
+        session_ending: &mut bool,
+        handler_context: &HandlerContext,
+    ) -> ControlFlow<(), ()> {
```

Comment:
This seems like a weird return type? This type is usually suggested for when you want the caller to have a choice of what to do. What made this necessary instead of just a `bool`?

Reply:
indeed. ;) seems i was too creative: 8962968bd8ffdab246c5454f4d7a5daa77dc0f7f